packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -116,7 +116,7 @@ class ScrollPositionWithSingleContext extends ScrollPosition implements ScrollAc
     super.beginActivity(newActivity);
     _currentDrag?.dispose();
     _currentDrag = null;
-    if (!activity.isScrolling)
+    if (!activity.isScrolling && !(activity is HoldScrollActivity))
```

Comment:
Checking the type with `is` is kinda bad practice and we try to avoid it in flutter. Instead, I'd suggest checking some kind of a property (that we may have to add) to determine if we should go idle (similarly to how `isScrolling` works).

Reply:
`foo is! Foo` instead of `!(foo is Foo)` would be easier to read.

Reply:
You are right. I'll see about adding a property to ScrollActivity and its subclasses. Perhaps `isIdle`? :)

Reply:
Or, would that not be strange, i.e. `IdleScrollActivity.isIdle`? Would it be ok to check type with `is` in this case for if it is an `IdleScrollActivity`?